### ISSUE TITLE
[question] How can an action be called from within a nodeView?

### ISSUE THREAD

**Issue by charlex** [ifiokjr/remirror#131]
## Description

I've successfully set up a react nodeView (see https://github.com/ifiokjr/remirror/issues/129), but I cannot figure out how to call an action from within the nodeView.

The problem I'm having is that the Remirror context is not found (`useRemirror` throws an error), so I cannot get the actions from the context.

## Checklist

- [x] I have read the [**contributing**][contributing] document.

**Comment by charlex** [comment]
Ah, I see when the nodeView is created the props `view`, `getPosition`, and `node` can be used to dispatch a transaction.

Pseudo code:

```jsx
const MyNodeView: FC<MyNodeViewProps> = ({ node, getPosition, view, children }) => {
  const { attrs } = node;

  return (
    <div>
      {children}
      <Button
        onClick={() => {
          const transaction = view.state.tr.setNodeMarkup(
            getPosition(),
            node.type,
            {
              ...attrs,
              'data-counter': attrs['data-counter'] + 1,
            }
          );

          view.dispatch(transaction);
        }}
      />
    </div>
  );
};
```

**Comment by charlex** [comment]
@ifiokjr do you think this is ideal way to perform updates from within a nodeView? It would be preferable in terms of DRY to move this logic to a method on the Extension, and have the command and the nodeView call this method, right?

**Comment by ifiokjr** [comment]
> The problem I'm having is that the Remirror context is not found (`useRemirror` throws an error), so I cannot get the actions from the context.

@charlex this is a very good point. It would be difficult to add the same context as `useRemirror` but I see no reason why `actions`, `manager`, `view` and `state` can't be added to a `usePortalContext` hook. 

Let me quickly test it out and see if it would work.

**Comment by charlex** [comment]
@ifiokjr thank you!

**Comment by ifiokjr** [comment]
@charlex I just realised it should already have access to the manager, via the `useRemirrorManager` hook. 

To get access to actions you can try:

```ts
const { manager } = useRemirrorManager();
const { actions } = manager.data;

actions.doSomething('awesome');
```

Let me know if it works.

**Comment by ifiokjr** [comment]
@charlex let me know if #132 is a good enough solution.

**Comment by charlex** [comment]
@ifiokjr your solution in the above comment satisfied my question! 

I'm attempting to understand What #132's `usePortalContext()` offers beyond the above mentioned `useRemirrorManager()`. Essentially this is a helper context to avoid having to mentally juggle which place to pull values from (e.g.: should I use props? should I use context?)?

**Comment by ifiokjr** [comment]
@charlex I think for now `useRemirrorManager` is sufficient. The main properties it lacks are the `getPositioner` and `getRootProps`.

**Comment by ifiokjr** [comment]
@charlex it turns out that `useRemirrorManager` will currently not work since the Portals were broken. I've fixed them in the branch I'm working on and once I've untangled the code I'll submit a PR. 


**For the curious:**

The issue is to do with the following line.

https://github.com/ifiokjr/remirror/blob/c2fe2de0b5cfc51963f3a27c6e454a3a8f33210a/%40remirror/react/src/components/node-view-portal.tsx#L31

Changing `renderFunction()` to `<renderFunction />` allows for hooks and context to be used from within the component.
